src/metrics/metrics.jl

```diff
@@ -684,35 +520,116 @@ maximum shelter volume possible.
+        RSV[scenarios=scen] .= _shelter_species_loop(X[scenarios=scen], nspecies, colony_vol, max_colony_vol, k_area)
+    end
 
+    # Sum over groups and size classes to estimate total shelter volume
+    # proportional to the theoretical maximum (per site)
+    RSV = dropdims(sum(RSV, dims=:species), dims=:species)
 
-if ccall(:jl_generating_output, Cint, ()) == 1
-    Base.precompile(Tuple{Metric{typeof(_relative_shelter_volume),NTuple{4,Symbol},String},Array{Float64,3},Vector{Float64},Vararg{Any}})   # time: 1.6421927
-    Base.precompile(Tuple{Metric{typeof(_relative_juveniles),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3}})   # time: 0.4322211
-    Base.precompile(Tuple{Metric{typeof(_absolute_shelter_volume),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3},Vector{Float64},Vararg{Any}})   # time: 0.3056465
-    Base.precompile(Tuple{Metric{typeof(_total_absolute_cover),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3},Vector{Float64}})   # time: 0.2397704
-    Base.precompile(Tuple{Metric{typeof(_relative_taxa_cover),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3}})   # time: 0.1171238
-    Base.precompile(Tuple{typeof(_shelter_species_loop),NamedDimsArray{(:timesteps, :species, :sites),Float64,3,Array{Float64,3}},Int64,Vector{Float64},Vector{Float64},Vector{Float64},Vector{Float64}})   # time: 0.0916976
-    Base.precompile(Tuple{typeof(_absolute_shelter_volume),NamedDimsArray{(:timesteps, :species, :sites),Float64,3,Array{Float64,3}},Vector{Float64},DataFrame})   # time: 0.0089155
-    Base.precompile(Tuple{typeof(_relative_shelter_volume),NamedDimsArray{(:timesteps, :species, :sites),Float64,3,Array{Float64,3}},Vector{Float64},Vector{Float64},DataFrame})   # time: 0.0080667
+    clamp!(RSV, 0.0, 1.0)
+    return RSV
+end
+function _relative_shelter_volume(rs::ResultSet)::YAXArray
+    return rs.outcomes[:relative_shelter_volume]
 end
+relative_shelter_volume = Metric(_relative_shelter_volume, (:timesteps, :sites, :scenarios))
+
+
+include("pareto.jl")
+include("ranks.jl")
+include("reef_indices.jl")
+include("scenario.jl")
+include("site_level.jl")
+include("temporal.jl")
+include("utils.jl")
+
+
+#if ccall(:jl_generating_output, Cint, ()) == 1
+#    Base.precompile(Tuple{Metric{typeof(_relative_shelter_volume),NTuple{4,Symbol},String},Array{Float64,3},Vector{Float64},Vararg{Any}})   # time: 1.6421927
+#    Base.precompile(Tuple{Metric{typeof(_relative_juveniles),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3}})   # time: 0.4322211
+#    Base.precompile(Tuple{Metric{typeof(_absolute_shelter_volume),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3},Vector{Float64},Vararg{Any}})   # time: 0.3056465
+#    Base.precompile(Tuple{Metric{typeof(_total_absolute_cover),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3},Vector{Float64}})   # time: 0.2397704
+#    Base.precompile(Tuple{Metric{typeof(_relative_taxa_cover),Tuple{Symbol,Symbol,Symbol},String},Array{Float64,3}})   # time: 0.1171238
```

Comment:
Should this be uncommented or removed?

Reply:
You know what? Go ahead and delete. We can regenerate those statements once everything settles down anyway.